pennylane/devices/experimental/device_api.py

```diff
@@ -237,6 +236,34 @@ def preprocess(
         * validation of configuration parameters
         * choosing a best gradient method and ``grad_on_execution`` value.
 
+        **Example**
+
+        All the transforms that are part of the preprocessing need to respect the transform contract defined here
+        :func:`pennylane.transforms.core.transform`: , you can find an example of a valid transform.
+
+        .. code-block:: python
+
+                @transform
+                def my_preprocessing_transform(tape: qml.tape.QuantumTape) -> (Sequence[qml.tape.QuantumTape], callable):
```

Comment:
I was thinking tape is the official user facing circuit representation? Maybe I am wrong? @trbromley

Reply:
Since this is developer-facing content rather than user-facing, I believe QS is the default internal unless queuing is required (I believe). QT is a QS, but inheriting also from the old annotated-queue class. Doing this for the base-class would allow both to be supported, rather than just QT, assuming all other functionality is equal.

Reply:
I'll let @trbromley make the call on this, but I'd also prepare to change this again later if needed.

Reply:
> Since this is developer-facing content rather than user-facing

While it's just us writing device preprocessing at this stage, we do expect external developer users to have to interact with this API in future (e.g., if they are a plugin developer).

Since we've geared the transform docs around tapes (rather than qscripts), and since I think our external users are more familiar with tapes, my preference is to use a tape in this instance.

As a later follow up, we can consider switching from tape to script in docstrings.